test_project/test_app/tests/test_storage.py

```diff
@@ -4,12 +4,80 @@
 
 from test_app.tests.base import AutocompleterTestCase
 from test_app.models import Stock, Indicator
-from test_app.autocompleters import StockAutocompleteProvider, CalcAutocompleteProvider
+from test_app.autocompleters import CalcAutocompleteProvider, FacetedStockAutocompleteProvider, StockAutocompleteProvider
 from test_app import calc_info
-from autocompleter import Autocompleter, registry, signal_registry
+from autocompleter import base, Autocompleter, registry, signal_registry
 from autocompleter import settings as auto_settings
 
 
+class FacetedStoringAndRemovingTestCase(AutocompleterTestCase):
+    fixtures = ['stock_test_data_small.json']
+
+    def test_store_facet_data(self):
+        """
+        Storing saves facet data
+        """
+        aapl = Stock.objects.get(symbol='AAPL')
+        provider = FacetedStockAutocompleteProvider(aapl)
+        provider.store()
+
+        provider_name = provider.get_provider_name()
+        # the FacetedStockAutocompleteProvider get_facets is set to ['sector']
+        facet_set_name = base.FACET_SET_BASE_NAME % (provider_name, 'sector', 'Technology',)
+        keys = self.redis.keys(facet_set_name)
+        self.assertEqual(len(keys), 1)
+
+        facet_map_name = base.FACET_MAP_BASE_NAME % (provider_name,)
+        keys = self.redis.keys(facet_map_name)
+        self.assertEqual(len(keys), 1)
+
+        facet_data = provider._deserialize_data(self.redis.hget(keys[0], aapl.id))
+        self.assertEqual(facet_data, [{'key': 'sector', 'value': aapl.sector}])
+
+    def test_second_store_removes_old_facet_data(self):
+        """
+        Store removes outdated facet data
+        """
+        aapl = Stock.objects.get(symbol='AAPL')
+        provider = FacetedStockAutocompleteProvider(aapl)
+        provider.store()
+
+        provider_name = provider.get_provider_name()
+        # the FacetedStockAutocompleteProvider get_facets is set to ['sector']
+        facet_set_name = base.FACET_SET_BASE_NAME % (provider_name, 'sector', 'Technology',)
+        keys = self.redis.keys(facet_set_name)
```

Comment:
while it's good to know that that set was created ... you want to do scard as well i think and see how many objects ids were in that set, no?